aten/src/ATen/core/PythonFallbackKernel.cpp

```diff
@@ -62,9 +62,13 @@ void pythonFallback(const c10::OperatorHandle& op, torch::jit::Stack* stack) {
   StashTLSOnEntryGuard stash_guard;
 
   // If Python Mode is active, use its PyInterpreter for dispatch
-  const auto& maybe_python_mode_state = at::impl::PythonModeTLS::get_state();
+  // need a copy, not a reference because we will set the state to none before
+  // dispatching
+  const auto maybe_python_mode_state = at::impl::PythonModeTLS::get_state();
   if (maybe_python_mode_state) {
+    at::impl::PythonModeTLS::set_state(nullptr);
     maybe_python_mode_state->pyinterpreter()->dispatch(op, stack, maybe_python_mode_state);
+    at::impl::PythonModeTLS::set_state(maybe_python_mode_state);
```

Comment:
To summarize... when we get to the __torch_dispatch__ , the mode is disabled?

1) Is there a way to turn the mode back on in torch_dispatch? (or do we not care about that at all?)
2) Does this need a RAII guard? If dispatch raises an error, then the mode is going to be off completely. (If so, this could be a test case)

Reply:
> To summarize... when we get to the torch_dispatch , the mode is disabled?

Yes--this part matches the torch_function version

> Is there a way to turn the mode back on in torch_dispatch? (or do we not care about that at all?)

Yes--this is actually similar to what will happen for nesting. In that case, the outer mode's torch_dispatch will end up getting wrapped in a mode call so it will be:

outer mode hits this call -> disable mode -> outer mode's torch dispatch wrapper -> outer mode enables inner mode (we can do this because the mode's disabled) -> call outer's torch_dispatch with inner mode enabled

> Does this need a RAII guard? If dispatch raises an error, then the mode is going to be off completely. (If so, this could be a test case)

Hmmm good point, I think so yes. There are already a lot of tests that raise errors though so I'm a little interested that we haven't already hit that

Reply:
You can copy paste some of the RAII above in this file to do that easily.

Reply:
RAII added

Reply:
Looks good